### PR TITLE
tree-wide: voidify unchecked snprintf calls or use snprintf_ok

### PR DESCRIPTION
**Pull request by bluca**
According to Coverity, 194 ouf of 227 times we check for snprintf return code.
Voidify the rest.

CID#1461512
CID#1461513
CID#1461514
CID#1461515
CID#1461516
CID#1461518
CID#1461519
CID#1461520
CID#1461522

### CODE REVIEW

File: man/hwdb-usb-device.c [hunk 1]
```diff
@@ -9,7 +9,7 @@ int print_usb_properties(uint16_t vid, uint16_t pid) {
   int r;
 
   /* Match this USB vendor and product ID combination */
-  snprintf(match, sizeof match, "usb:v%04Xp%04X", vid, pid);
+  (void) snprintf(match, sizeof match, "usb:v%04Xp%04X", vid, pid);
```
Comment (keszybz) [inline man/hwdb-usb-device.c]: This should be xsprintf (and `match` more explicitly sized).

File: src/basic/format-util.c [hunk 2]
```diff
@@ -15,9 +15,9 @@ char *format_ifname_full(int ifindex, char buf[static IF_NAMESIZE + 1], FormatIf
                 return NULL;
 
         if (FLAGS_SET(flag, FORMAT_IFNAME_IFINDEX_WITH_PERCENT))
-                snprintf(buf, IF_NAMESIZE + 1, "%%%d", ifindex);
+                (void) snprintf(buf, IF_NAMESIZE + 1, "%%%d", ifindex);
         else
-                snprintf(buf, IF_NAMESIZE + 1, "%d", ifindex);
+                (void) snprintf(buf, IF_NAMESIZE + 1, "%d", ifindex);
```
Comment (keszybz) [inline src/basic/format-util.c]: `assert(snprintf_ok(…))` ?

File: src/libsystemd/sd-bus/bus-match.c [hunk 3]
```diff
@@ -997,19 +997,19 @@ const char* bus_match_node_type_to_string(enum bus_match_node_type t, char buf[]
...
                 return buf;
 
         case BUS_MATCH_ARG_NAMESPACE ... BUS_MATCH_ARG_NAMESPACE_LAST:
-                snprintf(buf, l, "arg%inamespace", t - BUS_MATCH_ARG_NAMESPACE);
+                (void) snprintf(buf, l, "arg%inamespace", t - BUS_MATCH_ARG_NAMESPACE);
                 return buf;
 
         case BUS_MATCH_ARG_HAS ... BUS_MATCH_ARG_HAS_LAST:
-                snprintf(buf, l, "arg%ihas", t - BUS_MATCH_ARG_HAS);
+                (void) snprintf(buf, l, "arg%ihas", t - BUS_MATCH_ARG_HAS);
```
Comment (keszybz) [inline src/libsystemd/sd-bus/bus-match.c]: Ignoring the return value here seems wrong…

Comment (bluca) [reply src/libsystemd/sd-bus/bus-match.c]: moved to snprintf_ok

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**